src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -56,6 +57,11 @@ object FilterIndexRule
             findCoveringIndexes(filter, outputColumns, filterColumns)
           FilterIndexRanker.rank(spark, filter, candidateIndexes) match {
             case Some(index) =>
+              val useBucketSpec = if (HyperspaceConf.filterRuleBucketCheckEnabled(spark)) {
```

Comment:
Should we do a ratio? If we selected 180 / 200 buckets, it may not be good to fall back to use bucket spec since we are limiting parallelism further without pruning many buckets.